tmail-backend/blob/blobid-list/src/main/java/com/linagora/tmail/blob/blobid/list/SingleSaveBlobStoreDAO.java

```diff
@@ -25,19 +25,20 @@ public SingleSaveBlobStoreDAO(BlobStoreDAO blobStoreDAO,
 
     public Mono<Void> save(BucketName bucketName, BlobId blobId, byte[] data) {
         return Mono.from(blobIdList.isStored(blobId))
-                .filter(isStored -> Objects.equals(isStored, false))
-                .switchIfEmpty(Mono.error(new ObjectStoreException("Can not save duplicate blob when single save is enabled")))
-                .then(Mono.from(blobStoreDAO.save(bucketName, blobId, data))
-                        .then(Mono.from(blobIdList.store(blobId))))
+                .filter(isStored -> isStored.equals(false))
+                .switchIfEmpty(Mono.empty())
+                .then(Mono.from(blobIdList.store(blobId))
```

Comment:
Add a test you will see that you always store the blob.

Reply:
How can I write a test for this case? Both return `Mono<Void>` and BlobId is PK.